packages/core/src/transform/diagnostics/augmentation.ts

```diff
@@ -201,8 +210,8 @@ function checkImplicitAnyError(
 
     // This error may appear either on `<Foo />` or `{{foo}}`/`(foo)`
     let globalName =
-      sourceNode.type === 'ElementNode'
-        ? sourceNode.tag.split('.')[0]
+      sourceNode.type === 'ElementNameNode'
+        ? sourceNode.value.split('.')[0]
         : sourceNode.type === 'PathExpression' && sourceNode.head.type === 'VarHead'
```

Comment:
This kind of stuff should now work better, since `ElementNode`, `MustachesStatement` and `SubExpression` all have the same `path` property